core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -977,6 +977,20 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         listing.delete(classOf[LogInfo], log.logPath)
       }
     }
+    // Delete log files that don't exist in listing database and exceed the configured max age.
+    Option(fs.listStatus(new Path(logDir))).map(_.toSeq).getOrElse(Nil)
```

Comment:
I think the problem with this is - what would happen if new event logs are added into the event logging directory and the `cleanLogs` thread runs before the `checkForLogs` thread runs, then we end up deleting the newly added event logs. Is my understanding right ?